pkg/vdr/web/resolver.go

```diff
@@ -0,0 +1,74 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package web
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/hyperledger/aries-framework-go/pkg/doc/did"
+	"github.com/hyperledger/aries-framework-go/pkg/framework/aries/api/vdr"
+)
+
+// Read resolves a did:web did.
+func (v *VDR) Read(didID string, opts ...vdr.ResolveOpts) (*did.Doc, error) {
+	// apply resolve opts
+	docOpts := &vdr.ResolveDIDOpts{}
+
+	for _, opt := range opts {
+		opt(docOpts)
+	}
+
+	var customClient bool
```

Comment:
If you choose string instead of bool here then you don't need second switch for error message in sendResolveRequest() -  you can use clientType directly in your error message.

Reply:
good call😁

Reply:
I would not bother with this.

I would initialize `docOpts` with `&http.Client{}` and then let the user override if they want. There is no requirement to distinguish the client used in the logs.